lib/BuildSystem/BuildSystemFrontend.cpp

```diff
@@ -388,6 +424,8 @@ void BuildSystemFrontendDelegate::resetForBuild() {
   auto impl = static_cast<BuildSystemFrontendDelegateImpl*>(this->impl);
 
   impl->numFailedCommands = 0;
+  impl->numErrors = 0;
+  impl->setStatus(BuildSystemFrontendDelegateImpl::Status::Uninitialized);
```

Comment:
I don't think this is right... this will be the wrong value if the status has already been Initialized.

I think all this should do is reset to Initialized if the status is Cancelled.